### PR TITLE
Our capstone project

### PR DESCRIPTION
**Pull request by omarsalem7**
## What does this PR do?

This PR is raised to our capstone project

## Description of the tasks completed

✔️ Implemented all methods visible in the diagram
✔️ Implemented all associations (1-to-many relationships) and aggregations (parent-child relationships) between classes
✔️ Created each class from the [UML class diagram] in a separate .rb file
✔️ Added unit tests for all implemented methods
✔️ Created a main.rb file that will serve as the console app entry-point.
✔️ Preserved all data by saving collections in .json files
✔️ [Demo of project feature](https://drive.google.com/file/d/1U_j9imjUm3xOV_J4Wlzzw-Gp1Fsm5Q6n/view?usp=sharing)

## How should it be tested

Clone the repository
```
git clone https://github.com/omarsalem7/Catalog-of-things.git
```
Navigate to the folder
```
cd Catalog-of-things
```
Checkout the branch
```
git checkout dev
```

### CODE REVIEW

File: classes/modules/author_module.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      puts "#{index + 1}) ID: #{author['id']} | First name: #{author['first_name']} | Last name: #{author['last_name']}"
+    end
+  end
+
+  def add_to_store(object)
+    object = object.to_hash
+    @authors << object unless @authors.include?(object)
+    File.write(@storage_file, JSON[@authors])
+  end
+end
```
Comment (Emmyn5600) [inline classes/modules/author_module.rb]: Good job data is saved after quitting the app however

   * Make sure that the app does not crash if the .json file does not exist or is empty. 
   
   Please fix the issue that occur when a  `.json` file doesn't exist or is empty.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @BertrandConxy @Yerimah  and @omarsalem7 

Good job so far! I know you have worked hard on making this milestone. There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
      
## Required Changes :recycle:

- [ ] Check the comment under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!:clap::clap::clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/modules/author_module.rb (hunk 1 above)
Comment (BertrandConxy) [reply classes/modules/author_module.rb]: Thank you @Emmyn5600. The issue was solved. 👍

File: classes/modules/games_module.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+require_relative '../author'
+require_relative './author_module'
+require 'json'
+
+class GameModule
+  attr_accessor :games
+
+  def initialize
+    @storage_file = './data/games.json'
+    file = JSON.parse(File.read(@storage_file))
```
Comment (mmsesay) [inline classes/modules/games_module.rb]: - [ ] I can see you did solve the `No such file or directory` error for the `author_module.rb` file. However, `games_module.rb` file is still throwing an error is a file doesn't exist.

I will suggest you check this out. And try to solve the error for all modules that are making use of files.

See error screenshot |
:---------------------------:|
a1

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by AshakaE — APPROVED** [review]
Hi Team, 

Thanks for making the requested changes 👏 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**